# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Tuborg49/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

### My lane: Refresh / Content Opportunity Scoring

I chose the Refresh / Content Opportunity Scoring lane because I want to investigate which content pages should be reviewed first for refresh, improvement, protection, pruning, or monitoring. This is useful because a content team has limited time and cannot review every page equally. I want to use observable search and engagement signals from the starter dataset to help rank pages for human review. The goal is not to claim that a model can guarantee which page will improve, but to see whether the available data can support a useful and explainable review priority.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [13]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


### Decision

The decision is which content pages should be reviewed first for a possible refresh or other content action.

### Action

A content or SEO reviewer could use the ranked results to decide which pages to inspect first. The reviewer could then choose an appropriate action such as refreshing, improving, protecting, monitoring, or deciding that no change is needed.

### Cost of a wrong call

A false positive could cause the team to spend limited time improving a page that does not need attention. A false negative could cause the team to miss a page with a real opportunity or problem. Therefore, the ranking should support human review rather than automatically deciding what should be changed.

### Unit of analysis

The unit of analysis is one content page.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [14]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [15]:
import pandas as pd

url = url = "https://raw.githubusercontent.com/Tuborg49/flyrank-ml-internship/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Number of rows:", len(df))
print("Average 90-day impressions:", round(df["impressions_90d"].mean(), 2))
print("Percentage of pages with declining trend:",
      round((df["trend_direction"] == "down").mean() * 100, 2))


Number of rows: 30000
Average 90-day impressions: 5200.37
Percentage of pages with declining trend: 54.21


### What the starter data shows

The starter dataset contains 30,000 page-level records. The average number of 90-day impressions is 5,200.37, and 54.21% of pages have a declining trend. These numbers suggest that there is enough page-level variation to investigate how content pages could be prioritized for review. They do not prove that a refresh will improve a page, but they make the Refresh / Content Opportunity Scoring lane worth investigating over the next 7 weeks.

### What I can claim

I can say that the starter dataset contains 30,000 page-level records and includes useful signals such as 90-day impressions and trend direction. I can investigate whether these signals can help prioritize pages for human review.

I can also compare different ways of ranking pages and evaluate whether the rankings are useful for the decision.

### What I cannot claim

I cannot claim that a declining page will definitely improve after a refresh. I cannot claim that the data proves that a particular content change causes better search performance. I also cannot claim that the results will automatically generalize to every client or future dataset.

The purpose of this work is decision support and investigation, not predicting Google's decisions or automatically deciding what a content team should do.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What I can claim

I can say that the starter dataset contains 30,000 page-level records and includes useful signals such as 90-day impressions and trend direction. I can investigate whether these signals can help prioritize pages for human review.

I can also compare different ways of ranking pages and evaluate whether the rankings are useful for the decision.

### What I cannot claim

I cannot claim that a declining page will definitely improve after a refresh. I cannot claim that the data proves that a particular content change causes better search performance. I also cannot claim that the results will automatically generalize to every client or future dataset.

The purpose of this work is decision support and investigation, not predicting Google's decisions or automatically deciding what a content team should do.

In [16]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.